src/pybind/mgr/orchestrator/module.py

```diff
@@ -1099,7 +1099,8 @@ def _apply_misc(self,
                 specs.append(spec)
         else:
             placementspec = PlacementSpec.from_string(placement)
-            assert service_type
+            if not service_type:
```

Comment:
What happens if the service type provided is...  for example "fortnite"?
and 
What happens if I try "ceph orch apply mon 34567"?

Reply:
@jmolmo giving a service type that isn't a real service will still fail this check and print the error message. Service types are no longer strings and are now part of an an Enum ""ServiceType" class in the orchestrator module (https://github.com/ceph/ceph/blob/master/src/pybind/mgr/orchestrator/module.py#L40). It is impossible for service type here to be anything not listed in that class. If something else like "fortnite" is given it will become None and therefore fail this check. "ceph orch apply mon 34567" would run fine and create a mon service spec with a placement count of 34567. Whether we want to allow that many mons in the placement, if it's an issue at all, is different issue then what this PR is addressing

Reply:
what about using https://github.com/ceph/ceph/pull/39223/files#diff-91ff13f38df816b588aedff4193f02803771e1702b8dabd8f67b3e36d939e570R516-R523 ?